### ISSUE TITLE
ParameterModeInMethodCall rule problem

### ISSUE THREAD

**Issue by gquerret** [Riverside-Software/sonar-openedge#1160]
On behalf of MFe

When class contains overloaded versions of a method, one with `input-output` and the other with `output`, then `ParameterMode` rule incorrectly reports calls to this method.

```
class MyClass:

  method public void method01(input-output prm1 as char):
    //
  end method.

  method public void method01(output prm1 as char):
    //
  end method.

  method public void method02():
    var character xx.
    // Does not compile
    // this-object:method01(xx).
    this-object:method01(input-output xx). // Flagged by rule
    this-object:method01(output xx).
  end method.

end class.
```

Questionable design though.
